src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1547,14 +1547,15 @@ Base.vcat(dfs::AbstractDataFrame...;
                       AbstractVector{<:AbstractString}}=:setequal) =
     reduce(vcat, dfs; cols=cols)
 
-Base.reduce(::typeof(vcat),
-            dfs::Union{AbstractVector{<:AbstractDataFrame},
-                       Tuple{Vararg{AbstractDataFrame}}};
-            cols::Union{Symbol, AbstractVector{Symbol},
-                        AbstractVector{<:AbstractString}}=:setequal) =
-    _vcat([df for df in dfs if ncol(df) != 0]; cols=cols)
+function Base.reduce(::typeof(vcat),
+                     dfs::Union{AbstractVector{<:AbstractDataFrame},
+                                Tuple{Vararg{AbstractDataFrame}}};
```

Comment:
I think this should ensure that we do not overwrite the implementation from base for `reduce(vcat, ())`:
```suggestion
                                Tuple{AbstractDataFrame, Vararg{AbstractDataFrame}}};
```
(for vectors it is OK to accept 0-length vectors as they have `eltype`)